src/main/jbake/content/blog/2017/06/23/opentracing-appmetrics.adoc

```diff
@@ -0,0 +1,305 @@
+
+The example code includes a script that loops, randomly invoking the three REST endpoints provided by
+_ordermgr_. Once some example requests have been created, you can view the tracing dashboard:
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-06-23-traces.png[caption="Figure 2: ", title="Jaeger tracing dashboard"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-06-23-traces.png[caption="Figure 2: ", title="Jaeger tracing dashboard"]
+endif::[]
+
+Then you can select a specific trace instance and see further details:
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-06-23-trace.png[caption="Figure 3: ", title="Jaeger trace instance view"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-06-23-trace.png[caption="Figure 3: ", title="Jaeger trace instance view"]
+endif::[]
+
+This shows that the trace instance has three spans, the first representing the receipt of the `/buy` request on
+_ordermgr_, the second where _ordermgr_ is invoking _accountmgr_, and finally the _accountmgr_ receiving the
+`/hello` request. In this particular trace instance, the _accountmgr_ invocation has reported an error, indicated
+by the `error=true` tag.
+
+Now we will look at the Grafana dashboard to see what metrics have been reported from the OpenTracing
+instrumentation within the two services:
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-06-23-grafana-dashboard.png[caption="Figure 4: ", title="Grafana dashboard"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-06-23-grafana-dashboard.png[caption="Figure 4: ", title="Grafana dashboard"]
+endif::[]
+
+This dashboard includes three graphs, the first showing the number of spans created (i.e. span count) by
+our `sell()` method, and we can use it to track how many times this business operation has been executed.
+The second showing the average duration of the spans, and third showing the ratio between successful and
+erronous spans.
+
+The metrics reported by Prometheus are based on a range of labels - a metric exists for each unique combination
+of those labels.
+
+The standard labels included with the OpenTracing java-metrics project are: `operation`, `span.kind` and `error`.
+
+With this particular example, we also included the `transaction` label.
+
+However when the services are deployed to Kubernetes, the following additional labels are included for free: `pod`,
+`instance`, `service`, `job` and `namespace`.
+
+In our example Prometheus queries, we have ignored most of the Kubernetes added labels (except `service`) so that
+the metrics are aggregated across the specific pods, namespaces, etc. However, having these labels available means
+it is possible to segment the metrics in whatever way is required to analyse the data.
+
+When using the `java-metrics` project outside of Kubernetes, it is still possible to include the `service`
+label, however this would be configured when setting up the tracer.
+
```

Comment:
>however this would be configured when setting up the tracer.

Passive voice, so it's unclear who or what is doing the configuring.  Is it automatic?  Or manual?  If it's manual, revise as   "...you would configure this when setting up the tracer."